clinet/src/components/NamePasswordFrom/NamePasswordForm.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {useState} from "react";
+
+import "./NamePasswordForm.css";
+
+type Props = {
+    title: string;
+    action: string;
+    onSubmit: (name: string, password: string) => void;
+    alternative?: JSX.Element;
+};
+
+type ChangeEvent = React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>;
+
+const useFormInput = (initialState: string): [string, (event: ChangeEvent) => void] => {
+    const [value, setValue] = useState<string>(initialState);
+
+    const onChange = (e: ChangeEvent) => {
+        setValue(e.target.value);
+    };
+
+    return [value, onChange];
+};
+
+const NamePasswordForm: React.FunctionComponent<Props> = ({
```

Comment:
Spelling nitpick: `NamePasswordFrom` -> `NamePasswordForm`

(the parent dir name)